Added Flatpacks and Armour sets to blocklist

zukkiers opened the pull request:
Added most flatpacks to blocklist (excluding 5 that share exact name with their drink counterpart, if it possible please tell me I am interested)
Added all GE sets to blocklist
I feel these are not things that should be on the roll table.
Obviously up to you if you want to add them or not.

ChunkyAtlas commented:
Still contemplating if I want to remove flatpacks, also it would be be easy enough to switch from just using item name to item name and ID. I'll do that in the future when I get more time.

joshuagagne commented:
As much as it's frustrating me to roll flatpacks, it sort of gives unlocking construction a huge potential.  Honestly with this type of account, more rolls means you're less likely to 'never get a knife' because there are fewer ways to trigger a roll. Even the armour packs, they give a bit of a buff to "I rolled an unlock that completes a set! WOO, extra roll".   I kinda like that myself.  (not that I've unlocked a set or construction yet ;) )

zukkiers commented:
There is a around 80 flatpacks in the game so just over 2% of all rollable items. I'd argue construction already has huge unlocks with clockworks and toys. there are over 100 sets in the game and completing any of them is unlikely especially clue sets which there are the most of.
What I think is the most important reason they should be removed is that neither adds anything.
if you can make a flatpack you can just build it in your house. if you can make a set you already have the 4 items. there is a lot of random "junk" in the game but they have their own purpose in some way atleast by being either cosmetic or required for a quest. 
If nothing else I would really like for the player to have the option in the plugin to choose if they want to remove flatpacks and sets

zukkiers commented:
Changed the Pull request now it should add a checkbox to the configs if someone wants to enable the "gudi rule set" Where flatpacks, sets and poisoned weapons are not rollable I do not know how to make it so poison weapons are unusable till you roll the equivalent weapon poison so for now it's a player kept rule.
The code I put in is dumb but it should work, I almost entirely just cloned existing features/variables and added an extra check in the Refreshtradeable function.

If you want to move sets or flatpacks to the original blocklist, tell me and I can revert it. but I think this is a good solution for those who want to play with Gudi's ruleset.

ChunkyAtlas commented:
I'll meet you in the middle, have a toggle for all three not just one, (EG Remove Flatpaks, Remove Armor Sets, Remove Poison)
